src/sysinfo/nvml.cpp

```diff
@@ -14,27 +14,28 @@ namespace dxvk {
             return;
         }
 
-        m_nvmlInit_v2 = GetProcAddress<PFN_nvmlInit_v2>("nvmlInit_v2");
-        m_nvmlShutdown = GetProcAddress<PFN_nvmlShutdown>("nvmlShutdown");
-        m_nvmlErrorString = GetProcAddress<PFN_nvmlErrorString>("nvmlErrorString");
-        m_nvmlDeviceGetHandleByPciBusId_v2 = GetProcAddress<PFN_nvmlDeviceGetHandleByPciBusId_v2>("nvmlDeviceGetHandleByPciBusId_v2");
-        m_nvmlDeviceGetPciInfo_v3 = GetProcAddress<PFN_nvmlDeviceGetPciInfo_v3>("nvmlDeviceGetPciInfo_v3");
-        m_nvmlDeviceGetTemperature = GetProcAddress<PFN_nvmlDeviceGetTemperature>("nvmlDeviceGetTemperature");
-        m_nvmlDeviceGetUtilizationRates = GetProcAddress<PFN_nvmlDeviceGetUtilizationRates>("nvmlDeviceGetUtilizationRates");
-        m_nvmlDeviceGetVbiosVersion = GetProcAddress<PFN_nvmlDeviceGetVbiosVersion>("nvmlDeviceGetVbiosVersion");
-        m_nvmlDeviceGetPerformanceState = GetProcAddress<PFN_nvmlDeviceGetPerformanceState>("nvmlDeviceGetPerformanceState");
-        m_nvmlDeviceGetClockInfo = GetProcAddress<PFN_nvmlDeviceGetClockInfo>("nvmlDeviceGetClockInfo");
+#define GETPROCADDR(x) m_##x = GetProcAddress<PFN_##x>(#x)
```

Comment:
I'm torn about this macro, it does indeed remove boilerplate code, but not sure if a macro wrapping a template is a wise thing.

Reply:
I think it's justified because it makes you literally unable to query wrong method (by passing wrong name by accident) and it's not hidden away in some other header file so it's easy to verify it really does what you expect it to do.

But I can revert this change if you'd like.

Reply:
Yeah, I'm seeing the benefit. Lets do it the other way around, could you please apply the same magic to https://github.com/jp7677/dxvk-nvapi/blob/master/tests/nvapi_system.cpp#L102 and the `GetInstanceProcAddress` usages in  https://github.com/jp7677/dxvk-nvapi/blob/master/src/sysinfo/vulkan.cpp (also moving them back into the constructor). That way we do similar scoped magic everywhere when obtaining function pointers.

Reply:
As discussed, lets do this in a separate PR.